docker/base-deps/Dockerfile

```diff
@@ -47,18 +45,24 @@ RUN sudo apt-get update -y && sudo apt-get upgrade -y \
         rsync \
         netbase \
         openssh-client \
-        gnupg; fi) \
-    && wget \
-        --quiet "https://repo.anaconda.com/miniconda/Miniconda3-py37_23.1.0-1-Linux-${HOSTTYPE}.sh" \
-        -O /tmp/miniconda.sh \
-    && /bin/bash /tmp/miniconda.sh -b -u -p $HOME/anaconda3 \
+        gnupg; fi) 
+RUN (if [ "$PYTHON_VERSION" = "3.11" ]; \
+        then wget \
+            --quiet "https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-${HOSTTYPE}.sh" \
+            -O /tmp/miniconda.sh; \
+        else wget \
+            --quiet "https://repo.anaconda.com/miniconda/Miniconda3-py37_23.1.0-1-Linux-${HOSTTYPE}.sh" \
+            -O /tmp/miniconda.sh; \
+        fi;) 
+RUN /bin/bash /tmp/miniconda.sh -b -u -p $HOME/anaconda3 \
     && $HOME/anaconda3/bin/conda init \ 
     && echo 'export PATH=$HOME/anaconda3/bin:$PATH' >> /home/ray/.bashrc \
-    && rm /tmp/miniconda.sh \
-    && $HOME/anaconda3/bin/conda install -y \
-        # remove libffi pinning when fixed upstream #33299
-        libgcc-ng libffi=3.3 python=$PYTHON_VERSION \
-    && $HOME/anaconda3/bin/conda clean -y --all \
+    && rm /tmp/miniconda.sh 
+RUN (if [ "$PYTHON_VERSION" = "3.11" ]; \
+        then $HOME/anaconda3/bin/conda install -y libgcc-ng python=$PYTHON_VERSION; \
+        else $HOME/anaconda3/bin/conda install -y libgcc-ng libffi=3.3 python=$PYTHON_VERSION; \
+        fi;) 
```

Comment:
I think it was intentionally using one RUN to keep it as one docker layer.

If you need more fancy/complicated stuff, maybe start using `DOCKER_BUILDKIT=1` and use a full bash script.

Reply:
I used RUN << EOF without success, so I inline everything into one RUN call. My guess is DockerClient python API does not understand that format.